relay-server/src/services/store.rs

```diff
@@ -989,21 +994,130 @@ impl StoreService {
 
-        for mut log in logs.items {
-            log.organization_id = scoping.organization_id.value();
-            log.project_id = scoping.project_id.value();
-            log.retention_days = retention_days;
-            log.received = safe_timestamp(received_at);
+        for log in logs.items {
+            let timestamp_seconds = log.timestamp_nanos / 1_000_000_000;
+            let timestamp_nanos = log.timestamp_nanos % 1_000_000_000;
+            let item_id = u128::from_be_bytes(
+                *Uuid::new_v7(uuid::Timestamp::from_unix(
+                    uuid::NoContext,
+                    timestamp_seconds,
+                    timestamp_nanos as u32,
+                ))
+                .as_bytes(),
+            )
+            .to_le_bytes()
+            .to_vec();
+            let received_nanos = if let Some(received) = received_at.timestamp_nanos_opt() {
+                received % 1_000_000_000
+            } else {
+                0
+            };
+            let mut trace_item = TraceItem {
+                item_type: TraceItemType::Log.into(),
+                organization_id: scoping.organization_id.value(),
+                project_id: scoping.project_id.value(),
+                received: Some(Timestamp {
+                    seconds: safe_timestamp(received_at) as i64,
+                    nanos: received_nanos as i32,
+                }),
+                retention_days: retention_days.into(),
+                timestamp: Some(Timestamp {
+                    seconds: timestamp_seconds as i64,
+                    nanos: timestamp_nanos as i32,
+                }),
+                trace_id: log.trace_id.to_string(),
+                item_id,
+                attributes: Default::default(),
+                client_sample_rate: 1.0,
+                server_sample_rate: 1.0,
+            };
+
+            trace_item.attributes.insert(
+                "sentry.timestamp_precise".to_string(),
+                AnyValue {
+                    value: Some(Value::StringValue(log.timestamp_nanos.to_string())),
+                },
+            );
+            trace_item.attributes.insert(
+                "sentry.timestamp_nanos".to_string(),
+                AnyValue {
+                    value: Some(Value::StringValue(log.timestamp_nanos.to_string())),
+                },
+            );
+            trace_item.attributes.insert(
+                "sentry.severity_text".to_string(),
+                AnyValue {
+                    value: Some(Value::StringValue(
+                        log.severity_text.unwrap_or_else(|| "INFO".into()).into(),
+                    )),
+                },
+            );
+            trace_item.attributes.insert(
+                "sentry.severity_number".to_string(),
+                AnyValue {
+                    value: Some(Value::IntValue(
+                        log.severity_number.unwrap_or_default().into(),
+                    )),
+                },
+            );
```

Comment:
This looks like business/product logic, does it make sense to move that to the log normalization?